File: Backend/Travle.Model/Responses/MonthlyBookingPoint.cs
namespace Travle.Model.Responses
{
    /// <summary>
    /// One bar of the "bookings per month" chart: the count of bookings created in a given calendar
    /// month (UTC). The series always carries a continuous run of months (gaps filled with zero) so the
    /// chart has no missing columns. <see cref="Label"/> is a ready-to-render "Mon yyyy" caption.
    /// </summary>
    public class MonthlyBookingPoint
    {
        public int Year { get; set; }
        public int Month { get; set; }
        public int Count { get; set; }

        /// <summary>Short display caption, e.g. "Aug 2026".</summary>
        public string Label { get; set; } = string.Empty;
    }
}

File: UI/travle_core/lib/src/models/monthly_booking_point.dart
import 'package:json_annotation/json_annotation.dart';

part 'monthly_booking_point.g.dart';

/// One bar of a bookings-per-month chart (mirrors the backend `MonthlyBookingPoint`).
/// [label] is a ready-to-render "Mon yyyy" caption; the series carries a continuous
/// run of months with gaps filled by zero.
@JsonSerializable()
class MonthlyBookingPoint {
  MonthlyBookingPoint({
    required this.year,
    required this.month,
    required this.count,
    required this.label,
  });

  final int year;
  final int month;
  final int count;
  final String label;

  factory MonthlyBookingPoint.fromJson(Map<String, dynamic> json) =>
      _$MonthlyBookingPointFromJson(json);

  Map<String, dynamic> toJson() => _$MonthlyBookingPointToJson(this);
}
